airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/file_type_parser.py

```diff
@@ -4,13 +4,14 @@
 
 import logging
 from abc import ABC, abstractmethod
-from typing import Any, Dict, Iterable
+from typing import Any, Dict, Iterable, Mapping, Optional
 
 from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
 from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
 from airbyte_cdk.sources.file_based.remote_file import RemoteFile
 
-Schema = Dict[str, str]
+# Schema example: {"column1": {"type": "string"}}
+Schema = Mapping[str, Dict[str, Any]]
```

Comment:
Should we re-use `schema_helpers.SchemaType` here? If not, what is the difference between the two?

Reply:
I'm almost wondering if it would make sense to create a class and have business level behavior like `add_column(self, name, type)`. This would avoid the forever complexity of MyPy and Dict vs Mapping